### PR TITLE
feat: add route in workflow to forward declaration of health systems

### PR DESCRIPTION
**Pull request by Nil20**
Farajaland PR: https://github.com/opencrvs/opencrvs-farajaland/pull/981

### CODE REVIEW

File: packages/search/src/elasticsearch/utils.ts [hunk 1]
```diff
@@ -325,7 +342,23 @@ export const createStatusHistory = async (
...
+    : await (async () => {
+        const payload = getTokenPayload(authHeader)
+        const system = await getSystem(
+          { systemId: payload.sub },
+          { Authorization: authHeader }
+        )
+        return system
+      })()
+
+  if (typeof user.name !== 'object') user.name = [user.name]
```
Comment (rikukissa) [inline packages/search/src/elasticsearch/utils.ts:360]: Can you clarify this condition a little? Is this a difference between a system user and a regular user? Would 

```ts
const name = isSystemUser(user) ? ... : ...
```

be safer here than a mutation?

File: packages/search/src/elasticsearch/utils.ts [hunk 2]
```diff
@@ -354,7 +387,7 @@ export const createStatusHistory = async (
     rejectComment: body.rejectComment,
     operatorRole:
       // user could be a system as well and systems don't have role
-      user.role?.labels.find((label) => label.lang === 'en')?.label || '',
+      user.role?.labels.find((label: any) => label.lang === 'en')?.label || '',
```
Comment (rikukissa) [inline packages/search/src/elasticsearch/utils.ts:390]: Type seems to be missing here

File: packages/workflow/src/records/handler/eventNotificationHandler.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  const systemInfo = await getLoggedInPractitionerResource(token)
+
+  const savedTaskWithRegLastUser = addExtensionsToTask(task, [
+    {
+      url: 'http://opencrvs.org/specs/extension/regLastUser',
+      valueReference: {
+        reference: `Practitioner/${systemInfo.id}`
+      }
+    }
+  ]) as SavedTask
```
Comment (rikukissa) [inline packages/workflow/src/records/handler/eventNotificationHandler.ts:51]: Is there a way to get rid of the type casts? I don't think these should be necessary 🤔

File: packages/workflow/src/records/handler/eventNotificationHandler.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      ...savedBundle.entry.filter((e) => e.resource.resourceType !== 'Task'),
+      {
+        fullUrl: savedBundle.entry.find(
+          (e) => e.resource.resourceType === 'Task'
+        )?.fullUrl,
+        resource: savedTask
+      },
+      ...[office, officeLocation].map((r) => resourceToSavedBundleEntry(r))
+    ]
+  } as ValidRecord
```
Comment (rikukissa) [inline packages/workflow/src/records/handler/eventNotificationHandler.ts:102]: Same here

File: packages/search/src/elasticsearch/utils.ts (hunk 1 above)
Comment (Nil20) [reply packages/search/src/elasticsearch/utils.ts:361]: For a system user, `user.name` is a string where for a regular user `user.name` is 

`name: {
    use: string
    family: string
    given: string[]
  }[]
 `
 on which we use a find operation later. So thought, i could wrap it for a system's `user.name` in an array. I've added a comment here. But maybe there could be another way for this.

File: packages/workflow/src/records/handler/eventNotificationHandler.ts (hunk 3 above)
Comment (Nil20) [reply packages/workflow/src/records/handler/eventNotificationHandler.ts]: Removed

File: packages/search/src/elasticsearch/utils.ts (hunk 1 above)
Comment (Nil20) [reply packages/search/src/elasticsearch/utils.ts:361]: Don't need it really, user can actually be null for a system

File: packages/workflow/src/records/handler/eventNotificationHandler.ts [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  const responseBundle = await sendBundleToHearth(
+    savedBundleWithRegLastUserAndBusinessStatus
+  )
+  const savedBundle = toSavedBundle(
+    savedBundleWithRegLastUserAndBusinessStatus,
+    responseBundle
+  ) as ValidRecord
+
+  await indexBundle(savedBundle, token)
+  await auditEvent('sent-notification', savedBundle, token)
```
Comment (Zangetsu101) [inline packages/workflow/src/records/handler/eventNotificationHandler.ts:147]: We were probably handling the "sent-notification" event in "create-record" endpoint as well. We can remove that now

File: packages/workflow/src/records/handler/eventNotificationHandler.ts [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+    ]
+  }
+
+  const responseBundle = await sendBundleToHearth(
+    savedBundleWithRegLastUserAndBusinessStatus
+  )
+  const savedBundle = toSavedBundle(
+    savedBundleWithRegLastUserAndBusinessStatus,
+    responseBundle
+  ) as ValidRecord
```
Comment (Zangetsu101) [inline packages/workflow/src/records/handler/eventNotificationHandler.ts:144]:
```suggestion
  )
  const record = changeState(savedBundle, 'IN_PROGRESS')
```

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

**Review by Zangetsu101 — APPROVED** [review]

**Pull request merged**